docs/understanding-tracking-design/json-schema-type-casting-rules/index.md

```diff
@@ -0,0 +1,1378 @@
+</td>
+</tr>
+<tr>
+<td>
+
+  ```json
+{
+    "type": "string",
+    "format": "date"
+}
+```
+
+</td>
+<td>
+
+`DATE`
+
+</td>
+</tr>
+<tr>
+<td>
+
+  ```json
+{
+    "type": "boolean"
+}
+```
+
+</td>
+<td>
+
+`BOOLEAN`
+
+</td>
+</tr>
+<tr>
+<td>
+
+  ```json
+{
+    "type": "string"
+}
+```
+
+</td>
+<td>
+
+`STRING`
+
+</td>
+</tr>
+<tr>
+<td>
+
+  ```json
+{
+    "type": "integer",
+    "minimum": N,
+    "maximum": M
+}
+```
+
+- `M` &le; 2147483647
+- `N` &ge; -2147483648
+
+</td>
+<td>
+
+`INT`
+
+</td>
+</tr>
+<tr>
+<td>
+
+  ```json
+{
+    "type": "integer",
+    "minimum": N,
+    "maximum": M
+}
+```
+
+- `M` &le; 9223372036854775807
+- `N` &ge; -9223372036854775808
+
+</td>
+<td>
+
+`BIGINT`
+
+</td>
+</tr>
+<tr>
+<td>
+
+```json
+{
+    "type": "integer",
+    "minimum": N,
+    "maximum": M
+}
+```
+
+- `M` &gt;1e38-1
+- `N` &lt;-1e38
+
+</td>
+<td>
+
+`DECIMAL(38,0)`
+
+</td>
+</tr>
+<tr>
+<td>
+
+```json
+{
+    "type": "integer",
+    "minimum": N,
+    "maximum": M
+}
+```
+
+- `M` &lt; 1e38-1
+- `N` &gt;-1e38
+
+</td>
+<td>
+
+`DOUBLE`
+
+</td>
+</tr>
+<tr>
+<td>
+
+```json
+{
+    "type": "integer"
+}
+```
+
+</td>
+<td>
+
+`BIGINT`
+
+</td>
+</tr>
+<tr>
+<td>
+
+```json
+{
+    "type": "number", // OR ["number", "integer"]
+    "minimum": N,
+    "maximum": M,
+    "multipleOf": F
+}
+```
+
+- `M` &le; 2147483647
+- `N` &ge; -2147483648
+- `F` is integer
+
+</td>
+<td>
+
+`INT`
+
+</td>
+</tr>
+<tr>
+<td>
+
+```json
+{
+    "type": "number", // OR ["number", "integer"]
+    "minimum": N,
+    "maximum": M,
+    "multipleOf": F
+}
+```
+
+- `M` &le; 9223372036854775807
+- `N` &ge; -9223372036854775808
+- `F` is integer
+
+</td>
+<td>
+
+`BIGINT`
+
+</td>
+</tr>
+<tr>
+<td>
+
+```json
+{
+    "type": "number", // OR ["number", "integer"]
+    "minimum": N,
+    "maximum": M,
+    "multipleOf": F
+}
+```
+
+- `M` &gt; 1e38-1
+- `N` &lt; -1e38
+- `F` is integer
+
+</td>
+<td>
+
+`DECIMAL(38,0)`
+
+</td>
+</tr>
+<tr>
+<td>
+
+```json
+{
+    "type": "number", // OR ["number", "integer"]
+    "minimum": N,
+    "maximum": M,
+    "multipleOf": F
+}
+```
+
+- `M` &lt; 1e38-1
+- `N` &gt; -1e38
+- `F` is integer
+
+</td>
+<td>
+
+`DOUBLE`
+
+</td>
+</tr>
+<tr>
+<td>
+
+```json
+{
+    "type": "number", // OR ["number", "integer"]
+    "multipleOf": F
+}
+```
+
+- `F` is integer
+
+</td>
+<td>
+
+`BIGINT`
+
+</td>
+</tr>
+<tr>
+<td>
+
+```json
+{
+    "type": "number", // OR ["number", "integer"]
+    "minimum": N,
+    "maximum": M,
+    "multipleOf": F
+}
+```
+
+
+- `P` &le; 38, where `P` is maximum precision (total number of digits) of `M` and `N`, adjusted for scale (number of digits after the `.`) of `F`.
+- `P` Rounded up to `9`, `18` or `38`, e.g. it could only take one of those 3 values.
+
+<details>
+
+`P` = `MAX`(`M.precision` - `M.scale` + `F.scale`,  `N.precision` - `N.scale` + `F.scale`)
+
+`S` = `F.scale`
+
+</details>
+
+For example, `M=10.9999, N=-10, F=0.1` will be `DECIMAL(3,1)`. Calculation as follows:
```

Comment:
Could you put the example directly below the list and merge what’s now inside of `<details>` with the derivation for this example? So it would be something like:

---

```
* P ...
* P ...

For example, for ... it would be `DECIMAL(9, 1)`.

<details>
<summary>The formulas</summary>

P = ...
S = ...

In the example above, M is ..., N is ..., ...
The calculation is as follows:
...
```